From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fabian=20Gr=C3=BCnbichler?= <f.gruenbichler@proxmox.com>
Date: Fri, 19 Dec 2025 08:46:27 +0100
Subject: [PATCH] Revert "tcp: add receive queue awareness in
 tcp_rcv_space_adjust()"

This reverts commit ea33537d82921e71f852ea2ed985acc562125efe.
---
 include/linux/tcp.h  | 2 +-
 net/ipv4/tcp_input.c | 6 ++----
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/include/linux/tcp.h b/include/linux/tcp.h
index 57e478bfaef20369f5dba1cff540e52c9302ebf4..7faabde885b4e4ccc4a32c90f063b5c40ea89002 100644
--- a/include/linux/tcp.h
+++ b/include/linux/tcp.h
@@ -339,7 +339,7 @@ struct tcp_sock {
 	} rcv_rtt_est;
 /* Receiver queue space */
 	struct {
-		int	space;
+		u32	space;
 		u32	seq;
 		u64	time;
 	} rcvq_space;
diff --git a/net/ipv4/tcp_input.c b/net/ipv4/tcp_input.c
index c58d1fb4e97d1629c7b003527c509a50f3ed274b..42b1578210614f8551578ac6a36f67098e03b876 100644
--- a/net/ipv4/tcp_input.c
+++ b/net/ipv4/tcp_input.c
@@ -780,7 +780,8 @@ static void tcp_rcvbuf_grow(struct sock *sk)
 void tcp_rcv_space_adjust(struct sock *sk)
 {
 	struct tcp_sock *tp = tcp_sk(sk);
-	int time, inq, copied;
+	u32 copied;
+	int time;
 
 	trace_tcp_rcv_space_adjust(sk);
 
@@ -791,9 +792,6 @@ void tcp_rcv_space_adjust(struct sock *sk)
 
 	/* Number of bytes copied to user in last RTT */
 	copied = tp->copied_seq - tp->rcvq_space.seq;
-	/* Number of bytes in receive queue. */
-	inq = tp->rcv_nxt - tp->copied_seq;
-	copied -= inq;
 	if (copied <= tp->rcvq_space.space)
 		goto new_measure;
 
